### PR TITLE
[#6622] Add test for closeAllL1Desc (main)

### PR DESCRIPTION
**Pull request by alanking**
Adds a test for the change which address #6622

Companion PR to #6623

---

I tried to add a test for PREP but found that `msiExecCmd` does not work in PREP right now. Instead of calling across to a native rule, I think that the current test is sufficient as it is.

### CODE REVIEW

File: scripts/irods/test/rule_texts_for_tests.py [hunk 1]
```diff
@@ -606,6 +606,25 @@
...
+'''
+rule_texts['irods_rule_engine_plugin-irods_rule_language']['Test_Rulebase']['test_msiExecCmd_closeAllL1Desc__issue_6623_pep'] ='''
+pep_resource_close_post(*inst, *ctx, *out) {{
+    msiGetSystemTime(*time_str, '');
+    msiAddKeyVal(*key_val_pair,"{attr}","*time_str");
+    msiAssociateKeyValuePairsToObj(*key_val_pair,"{logical_path}","-d");
+    # Sleep to ensure that the actual close doesn't occur in the same second
+    msiSleep("1", "0");
+}}
+'''
```
Comment (korydraughn) [inline scripts/irods/test/rule_texts_for_tests.py:627]: Consider moving these into their respsective test.

Having part of the test logic in a separate file introduces more hurdles to understanding the test.

If you see benefits to continuing to use this file, I'd be interested in hearing them.

Comment (alanking) [reply scripts/irods/test/rule_texts_for_tests.py:627]: I created #6677 to remove this file. Might as well just go ahead and do it. ;p

**Comment by alanking** [comment]
I have confirmed that all the tests pass in isolation but I'm running it through one more time to be sure.

**Comment by alanking** [comment]
For some reason, `test_stacktrace` falls over when run along with the rest of the suite, but it passes when run individually. I think this may be a fluke, so I'm going to say the tests are passing.

**Comment by trel** [comment]
sounds like a cleanup situation in the test prior to test_stacktrace...

**Comment by korydraughn** [comment]
Well, the test is relying on the log file so I agree it is likely a fluke/timing issue. We need a better way to confirm these things.

**Comment by alanking** [comment]
One test suite failed for PREP because it is only implemented for the native rule language. I have skipped it and referenced #4094. All the rest of the tests passed.

**Comment by alanking** [comment]
Finally made Codacy happy, so I'm gonna run tests just one more time. :)

**Comment by alanking** [comment]
All tests passed

**Review by korydraughn — APPROVED** [review]
Let's add the #'s.

**Pull request merged**